ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -1301,6 +1318,237 @@ static int ompi_osc_rdma_component_select (struct ompi_win_t *win, void **base,
 }
 
 
+
+
+static int ompi_osc_rdma_component_iselect (struct ompi_win_t *win, void **base, size_t size,
+                                            int disp_unit, struct ompi_communicator_t *comm, 
+                                            struct ompi_communicator_t **newcomm,
+                                            struct opal_info_t *info, int flavor,
+                                            struct ompi_request_t **request, int *model)
+{
+    int world_size = ompi_comm_size (comm);
+    int ret;
+    ompi_communicator_t *ncomm;
+    ompi_request_t *req;
+
+    /* the osc/sm component is the exclusive provider for support for shared
+     * memory windows */
+    if (MPI_WIN_FLAVOR_SHARED == flavor) {
+        return OMPI_ERR_NOT_SUPPORTED;
+    }
+
+    /* dup the communicator */
+    ret = ompi_comm_idup(comm, &ncomm, &req);
+    if(OMPI_SUCCESS != ret){
+        ompi_osc_rdma_free (win);
+        return ret;
+    }
+
+    *request = req;
+    *newcomm = ncomm;
+
+    return OMPI_SUCCESS;
+}
+
+
+/* This is expected to be called only when idup is complete, i.e.ompi_osc_rdma_component_iselect */
+static int ompi_osc_rdma_component_complete_iselect(struct ompi_win_t *win, void **base,
```

Comment:
A lot of this `complete_iselect` function can be moved into the `iselect` function because there are a lot of local-only actions here that don't depend on the duplication of the communicator.

Reply:
That's right. The only reason I decided to move it to `complete_iselect` is that all operations requiring `module`  happen within a single function.

Reply:
I'll put this on the agenda for discussion.